packages/adapter-node/src/index.js

```diff
@@ -1,15 +1,42 @@
 import { handler } from 'HANDLER';
 import { env } from 'ENV';
 import polka from 'polka';
+import http from 'node:http';
+import https from 'node:https';
+import http2 from 'node:http2';
+import fs from 'node:fs';
 
 export const path = env('SOCKET_PATH', false);
 export const host = env('HOST', '0.0.0.0');
 export const port = env('PORT', !path && '3000');
+export const certPath = env('CERT_PATH', false);
+export const certKeyPath = env('CERT_KEY_PATH', false);
+export const httpsPort = env('HTTPS_PORT', !path && '3001');
+export const onlyHttps = env('ONLY_HTTPS', false);
+export const noHttp2 = env('NO_HTTP2', false);
 
-const server = polka().use(handler);
+const app = polka().use(handler);
 
-server.listen({ path, host, port }, () => {
-	console.log(`Listening on ${path ? path : host + ':' + port}`);
-});
+if (!onlyHttps) {
+	// TODO Remove the `@ts-expect-error`s below once https://github.com/lukeed/polka/issues/194 is fixed
+	// @ts-expect-error
```

Comment:
Pretty sure you can bypass the ts-expect-errors by creating `server` via native modules (as you are) and then passing through `options.server` to Polka. The type errors are there cuz of Node's complex types.

```js
let server = certPath && certKeyPath
  ? (noHttp2 ? https.createServer({ ... }) : http2.createServer({ ... })
  : http.createServer();

let app = polka({ server });
```

Reply:
@lukeed This was brought up and I explained why I decided not to do it [here](https://github.com/sveltejs/kit/pull/10183?notification_referrer_id=NT_kwDOAZTGrbM2ODAyNzc1MTgzOjI2NTI3NDA1#issuecomment-1596199665).

Reply:
I saw but it's invalid because there's only 1 Polka instance & 1 server needed

Reply:
@lukeed Not that it matters anymore but that's incorrect, you could have 2 servers, a plain HTTP one and an HTTPS one; hence the intention to reuse the Polka instance.